pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java

```diff
@@ -170,16 +170,14 @@ public static void decodeRawGroupKey(long rawGroupKey, int[] cardinalities, int[
    */
   @Override
   public int generateKeyForDocId(int docId) {
+    long rawKey = 0;
+
     for (int i = 0; i < _groupByColumns.length; i++) {
-      if (_isSingleValueGroupByColumn[i]) {
-        _singleValIterators[i].skipTo(docId);
-        _reusableGroupByValuesArray[i] = _singleValIterators[i].nextIntVal();
-      } else {
-        throw new RuntimeException("GroupBy on multi-valued columns not supported.");
```

Comment:
Single group key and multi group keys share this class, and each of them calls different functions. This one is only called by single group key situation.